Update security.md (#231)
NilPointer-Software/bdfd-wiki: src/resources/security.md

This article will share tips about how you can keep your bot and account safe.

### Sharing Tokens and Passwords

> **Do not share token(s) with anyone**. This includes both bot and regular user account tokens. Sharing your bot token with someone (or posting it publicly) grants them full-unlimited access to your editing bot. Sharing your user account token with someone (or posting it publicly) allows them to have full access to your account (even if they don't have your password or email). Once someone has your account or bot's token, there is a high chance of it being used for malicious purposes. For example, stealing personal info, spreading scams, modifying your bot to nuke/raid servers.

Reviewer: ```suggestion
**Do not share token(s) with anyone**. This includes both bot and regular user account tokens. Sharing your bot token with someone (or posting it publicly) will grant them full access to edit your bot. While, sharing your user account token with someone (or posting it publicly) will allow them to have full access to your account (even if they don't have your password or email). Once someone has your account or bot's token, there is a high chance of it being used for malicious purposes. For example, stealing personal info, spreading scams, modifying your bot to nuke/raid servers etc.
```

Reviewer 2: No need for such a long line.
```suggestion
**Do not share token(s) with anyone**. This includes both bot and regular user account tokens. Sharing your bot token with someone (or posting it publicly) will grant them full access to edit your bot.
While sharing your user account token with someone (or posting it publicly) will allow them to have full access to your account (even if they don't have your password or email).
Once someone has your account or bot's token, there is a high chance of it being used for malicious purposes. For example, stealing personal info, spreading scams, modifying your bot to nuke/raid servers, etc.
```

> In the event that your **bot's token** is shared, the only thing you can do is regenerate it, but by then, the damage has most likely been done. In the event you share your **account token** but still have access to your account, you can regenerate your token by changing your password. If you don't, you will need to contact Discord support for an optimal solution.

Reviewer: ```suggestion
In an event, if your **bot's token** is shared, the only thing you can do to secure it, is to regenerate a new bot token but by then, the damage has most likely been done. In the case of an **user account token**, if you still have access to your account, regenerate your token by changing your password. If you cannot or don't have access to it anymore, you will need to contact [Discord support](https://support.discord.com/hc/en-us/requests/new) for an optimal solution.
```

Reviewer 2: This line is quite long, it should be split.
```suggestion
In the event, that your **bot's token** is shared, the only thing you can do to secure it is to regenerate the bot's token. But by then, the damage has most likely been done.
In the case of a **user account token**, if you still have access to your account, regenerate your token by changing your password. If you cannot or don't have access to it anymore, you will need to contact [Discord support](https://support.discord.com/hc/en-us/requests/new) for an optimal solution.
```

Passwords, like tokens, should not be shared. If, however, you accidentally share your account password, you should change it as soon as possible.

> 📝 If your account is hacked, you should [contact Discord](https://support.discord.com/hc/requests/new) for further assistance.

### Account 2FA
Bot owners should consider enabling two-factor authentication on their accounts. [Learn more about 2FA and why it's essential for bot owners](./2FA.md).

### Sessions
Discord recently added the ability to see all your current sessions and their respective locations.

![image](https://user-images.githubusercontent.com/116581988/199214548-911dc935-3c52-43cc-b0d7-74d18e283903.png)

If you see a device or location that you haven't authorised, you can log out of that particular device by pressing the 'X' button or all known devices by clicking the button at the bottom of the page. This will log out those sessions, invalidating the tokens. 

> ![image](https://user-images.githubusercontent.com/116581988/199320324-fd93384f-31e1-4a50-98c3-f4f875b91536.png)

Reviewer: ```suggestion
![image](https://user-images.githubusercontent.com/116581988/199320324-fd93384f-31e1-4a50-98c3-f4f875b91536.png)\
```

> 

Reviewer: ```suggestion
```

> ![image](https://user-images.githubusercontent.com/116581988/199320198-e856f6f2-47dc-4dda-bd8c-809eb34bcfdb.png)

Reviewer: ```suggestion
![image](https://user-images.githubusercontent.com/116581988/199320198-e856f6f2-47dc-4dda-bd8c-809eb34bcfdb.png)\
```

> 

Reviewer: ```suggestion
```

While sharing your user account token with someone (or posting it publicly) will allow them to have full access to your account (even if they don't have your password or email).
Once someone has your account or bot's token, there is a high chance of it being used for malicious purposes. For example, stealing personal info, spreading scams, modifying your bot to nuke/raid servers, etc.

In the event, that your **bot's token** is shared, the only thing you can do to secure it is to regenerate the bot's token. But by then, the damage has most likely been done.
In the case of a **user account token**, if you still have access to your account, regenerate your token by changing your password. If you cannot or don't have access to it anymore, you will need to contact [Discord support](https://support.discord.com/hc/en-us/requests/new) for an optimal solution.

Passwords, like tokens, should not be shared. If, however, you accidentally share your account password, you should change it as soon as possible.

> > 📝 If your account is hacked, you should [contact Discord](https://support.discord.com/hc/requests/new) for further assistance.

Reviewer: ```suggestion
> 📌 If your account is hacked, you should [contact Discord](https://support.discord.com/hc/requests/new) for further assistance.
```
